cabal-install/src/Distribution/Client/ProjectPlanning.hs

```diff
@@ -1024,13 +1024,13 @@ planPackages verbosity comp platform solver SolverSettings{..}
             | (pc, src) <- solverSettingConstraints ]
 
       . addPreferences
-          -- enable stanza preference where the user did not specify
+          -- enable stanza preference unilaterally, even when the user asked to enable as well, to help hint the solver.
```

Comment:
to make it clearer (future readers won't have the previous version of this code as context):
  -- enable stanza preference unilaterally, regardless if the user asked
  -- accordingly or expressed no preference, to help hint the solver